pkg/storage/tpr/install_types_test.go

```diff
@@ -0,0 +1,171 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tpr
+
+import (
+	"errors"
+	"strconv"
+	"strings"
+	"testing"
+
+	v1beta1 "k8s.io/kubernetes/pkg/apis/extensions/v1beta1"
+
+	kubeclientfake "k8s.io/kubernetes/pkg/client/clientset_generated/release_1_5/fake"
+	"k8s.io/kubernetes/pkg/client/testing/core"
+	"k8s.io/kubernetes/pkg/runtime"
+)
+
+var (
+	fakeClientset   *kubeclientfake.Clientset
+	getCallCount    int
+	createCallCount int
+	createCallArgs  []string
+	getCallArgs     []string
+)
+
+func setup(getFn, createFn func(core.Action) (bool, runtime.Object, error)) {
+	getCallCount = 0
+	createCallCount = 0
+	getCallArgs = []string{}
+	createCallArgs = []string{}
+
+	fakeClientset = &kubeclientfake.Clientset{}
+
+	fakeClientset.AddReactor("get", "thirdpartyresources", getFn)
+
+	fakeClientset.AddReactor("create", "thirdpartyresources", createFn)
+}
+
+//make sure all resources types are installed
+func TestInstallTypesAllResources(t *testing.T) {
+	setup(
+		func(core.Action) (bool, runtime.Object, error) {
+			getCallCount++
+			if getCallCount > len(thirdPartyResources) {
+				return true, &v1beta1.ThirdPartyResource{}, nil
+			}
+
+			return true, nil, errors.New("Resource not found")
+		},
+		func(core.Action) (bool, runtime.Object, error) {
+			createCallCount++
+			return true, nil, nil
+		},
+	)
+
+	installer := NewInstaller(fakeClientset.Extensions().ThirdPartyResources())
+	installer.InstallTypes()
+
+	if createCallCount != len(thirdPartyResources) {
+		t.Errorf("Error: The number of Third Party Resources installed is not 4")
```

Comment:
Consider changing this to be more prescriptive and remove Error from the log, I think it already gets added there by t.Error?
Basically it would be good to have what the actual count was, so something like:
"Expected %d Third Party Resources created by was %d", len(thirdPartyResources), createCallCount)

Or something like that.